packages/internals/src/client/getClientEngineType.ts

```diff
@@ -3,28 +3,36 @@ import type { GeneratorConfig } from '@prisma/generator-helper'
 export enum ClientEngineType {
   Library = 'library',
   Binary = 'binary',
+  Wasm = 'wasm',
 }
 
 export const DEFAULT_CLIENT_ENGINE_TYPE = ClientEngineType.Library
 
 export function getClientEngineType(generatorConfig?: GeneratorConfig): ClientEngineType {
-  const engineTypeFromEnvVar = getEngineTypeFromEnvVar()
+  const engineTypeFromEnvVar = getEngineTypeFromEnvVar(generatorConfig)
   if (engineTypeFromEnvVar) return engineTypeFromEnvVar
   if (generatorConfig?.config.engineType === ClientEngineType.Library) {
     return ClientEngineType.Library
   } else if (generatorConfig?.config.engineType === ClientEngineType.Binary) {
     return ClientEngineType.Binary
+  } else if (
+    generatorConfig?.config.engineType === ClientEngineType.Wasm &&
+    generatorConfig?.previewFeatures.includes('driverAdapters')
+  ) {
+    return ClientEngineType.Wasm
   } else {
     return DEFAULT_CLIENT_ENGINE_TYPE
   }
 }
 
-function getEngineTypeFromEnvVar() {
+function getEngineTypeFromEnvVar(generatorConfig?: GeneratorConfig) {
   const engineType = process.env.PRISMA_CLIENT_ENGINE_TYPE
   if (engineType === ClientEngineType.Library) {
     return ClientEngineType.Library
   } else if (engineType === ClientEngineType.Binary) {
     return ClientEngineType.Binary
+  } else if (engineType === ClientEngineType.Wasm && generatorConfig?.previewFeatures.includes('driverAdapters')) {
```

Comment:
Similar to the previuous comment, I feel like engineType=wasm witout driver adapters enabled should cause explicit validation error, rather than ignoring env var.